opds.py

```diff
@@ -1151,9 +1176,34 @@ def _make_entry_xml(self, work, license_pool, edition, identifier):
 
         return entry
 
-    def add_breadcrumbs(self, lane, annotator, include_lane=False):
-        """Add list of ancestor links in a breadcrumbs element."""
+
+    CURRENT_ENTRYPOINT_ATTRIBUTE = "{%s}entryPoint" % AtomFeed.SIMPLIFIED_NS
+
+    def show_current_entrypoint(self, entrypoint):
+        """Annotate this given feed with a simplified:entryPoint
+        attribute pointing to the current entrypoint's TYPE_URI.
+
+        This gives clients an overall picture of the type of works in
+        the feed, and a way to distinguish between one EntryPoint
+        and another.
+
+        :param entrypoint: An EntryPoint.
+        """
+        if not entrypoint:
+            return
+
+        if not entrypoint.URI:
+            return
+        self.feed.attrib[self.CURRENT_ENTRYPOINT_ATTRIBUTE] = entrypoint.URI
+
+    def add_breadcrumbs(self, lane, include_lane=False, entrypoint=None):
+        """Add list of ancestor links in a breadcrumbs element.
+
+        TODO: The switchover from "no entry point" to "entry point" needs
+        its own breadcrumb link.
```

Comment:
For some reason I thought "no entry point" would use a default entry point.

Reply:
When you visit the lane, you are assigned a default entry point for display purposes, but for breadcrumbs, "lane X" needs to come before "lane X but with entry point Y selected", even though they may be the same.